File: scatterplot.R
# Install and load the necessary library
if (!require(ggplot2)) {
  install.packages("ggplot2")
  library(ggplot2)
} else {
  library(ggplot2)
}

# Load the CSV data into the 'penglings' data frame
penglings <- read.csv("penglings.csv")

# Convert necessary columns to numeric if needed
penglings$flipper_length_mm <- as.numeric(penglings$flipper_length_mm)
penglings$body_mass_g <- as.numeric(penglings$body_mass_g)

# Create the scatter plot
plot <- ggplot(penglings, aes(x = flipper_length_mm, y = body_mass_g, color = species, 
                      size = bill_length_mm)) +
  geom_point(alpha = 0.8, shape = 16, stroke = 1.5, position = "jitter") +  
  # Add jitter to spread points
  scale_color_manual(values = c("Adelie" = "orange", "Chinstrap" = "purple", 
                                "Gentoo" = "#008080")) +  
  # Using hex code for teal
  scale_size_continuous(range = c(3, 12)) +  
  # Increased range for size
  scale_x_continuous(limits = c(170, 235), breaks = seq(170, 235, by = 5)) +  
  # Set x-axis limits and add breaks
  scale_y_continuous(limits = c(2500, 6500), breaks = seq(2500, 6500, by = 500)) +  
  # Set y-axis limits and add breaks
  theme_minimal() +
  theme(
    panel.grid.major = element_line(color = "white", size = 0.5),
    panel.grid.minor = element_line(color = "white", size = 0.5),
    panel.background = element_rect(fill = "lightgrey"),
    legend.position = "top",  
    # Moves the legend to the top
    axis.text.x = element_text(size = 10, angle = 45, hjust = 1),  
    # Angle and size for x-axis labels
    axis.ticks.x = element_line(color = "black", size = 0.5),  
    # Customize tick marks
    axis.line.x = element_line(color = "black"),  
    # Ensure the x-axis line is visible
    axis.text.y = element_text(size = 10)  
    # Adjust y-axis text size
  ) +
  labs(
    title = "Penguin Data Scatter Plot",
    x = "Flipper Length (mm)",
    y = "Body Mass (g)",
    color = "Species",  
    # Corrected legend title for species
    size = "Bill Length (mm)"  
  

  )

# Display the plot
print(plot)

File: scatter_plot.py
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

# Load CSV file (Ensure the file path is correct)
df = pd.read_csv("penglings.csv")

# Print column names for debugging
print(df.columns)

# Fix column names if necessary
x_column = "flipper_length_mm"  # Change to match your CSV file
y_column = "body_mass_g"
hue_column = "species" 
size_column = "bill_length_mm"

# Ensure these column names exist in the dataset
if x_column in df.columns and y_column in df.columns:
    plt.figure(figsize=(10, 6))
    sns.scatterplot(
        data=df,
        x=x_column,
        y=y_column,
        hue=hue_column,
        size=size_column,
        palette="viridis",
        edgecolor="black",
        alpha=0.7
    )

    # Format background and grid
    plt.grid(color='white', linestyle='-', linewidth=0.5)
    plt.gca().set_facecolor("lightgrey")

    # Labels and title
    plt.xlabel(x_column, fontsize=12)
    plt.ylabel(y_column, fontsize=12)
    plt.title("Penguin Data Scatter Plot", fontsize=14)

    # Move legend to the right
    plt.legend(loc='center left', bbox_to_anchor=(1, 0.5), title="Species")

    plt.show()
else:
    print("Error: Column names do not match the dataset. Check df.columns output.")
